newrelic/resource_newrelic_obfuscation_expression.go

```diff
@@ -0,0 +1,200 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/newrelic/newrelic-client-go/v2/newrelic"
+	_ "github.com/newrelic/newrelic-client-go/v2/newrelic"
+	nrErrors "github.com/newrelic/newrelic-client-go/v2/pkg/errors"
+	"github.com/newrelic/newrelic-client-go/v2/pkg/logconfigurations"
+)
+
+//
+func resourceNewRelicObfuscationExpression() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNewRelicObfuscationExpressionCreate,
+		ReadContext:   resourceNewRelicObfuscationExpressionRead,
+		UpdateContext: resourceNewRelicObfuscationExpressionUpdate,
+		DeleteContext: resourceNewRelicObfuscationExpressionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:        schema.TypeInt,
+				Description: "The account id associated with the obfuscation expression.",
+				Computed:    true,
+				Optional:    true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Description of expression.",
+				Optional:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Name of expression.",
+				Required:    true,
+			},
+			"regex": {
+				Type:        schema.TypeString,
+				Description: "Regex of expression.",
+				Required:    true,
+			},
+			"expression_id": {
+				Type:        schema.TypeInt,
+				Description: "Expression Id.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+//Create the obfuscation expression
+func resourceNewRelicObfuscationExpressionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+
+	accountID := selectAccountID(providerConfig, d)
+
+	createInput := logconfigurations.LogConfigurationsCreateObfuscationExpressionInput{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		Regex:       d.Get("regex").(string),
+	}
+
+	created, err := client.Logconfigurations.LogConfigurationsCreateObfuscationExpressionWithContext(ctx, accountID, createInput)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if created == nil {
+		return diag.Errorf("err: obfuscation expression create result wasn't returned or expression was not created.")
+	}
+
+	expressionID := created.ID
+
+	d.SetId(expressionID)
+
+	return resourceNewRelicObfuscationExpressionRead(ctx, d, meta)
+}
+
+//Read the obfuscation expression
+func resourceNewRelicObfuscationExpressionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	accountID := selectAccountID(providerConfig, d)
+
+	expressionID := d.Id()
+	expression, err := getObfuscationExpressionByID(ctx, client, accountID, expressionID)
+
+	if err != nil {
+		if _, ok := err.(*nrErrors.NotFound); ok {
```

Comment:
Does this work when you create an obfuscation expression from Terraform then delete in the UI? Terraform should try to recreate the resource.

Reply:
Thanks for pointing it out, made the required changes for it to work.